app/models/annotation/AnnotationService.scala

```diff
@@ -402,6 +403,22 @@ class AnnotationService @Inject()(annotationInformationProvider: AnnotationInfor
       zip <- createZip(nmlsAndNames, zipFileName)
     } yield zip
 
+  def listedAnnotationsFor(userTeams: List[ObjectId])(implicit ctx: DBAccessContext) =
+    for {
+      listedAnnotationIds <- listedAnnotationsDAO.findAllListedForTeams(userTeams)
+      annotations <- Fox.serialCombined(listedAnnotationIds)(annotationDAO.findOne(_)(GlobalAccessContext))
```

Comment:
I’m afraid this will become very slow once there are many annotations visible for the user, can this be combined into a single query? … then again, if this is complex, maybe we can do that later once it becomes a problem.